osf/models/preprint.py

```diff
@@ -925,6 +979,298 @@ def create_waterbutler_log(self, auth, action, payload):
             params=params
         )
 
+    def update_has_coi(self, auth: tuple, has_coi: bool, log: bool = True, save: bool = True):
+        """
+        This method sets the field `has_coi` to indicate if there's a conflict interest statement for this preprint
+        and logs that change.
+
+        :param auth: Tuple ('username', 'password')
+        :param has_coi: Boolean represents if a user has a conflict of interest statement available.
+        :param log: Boolean should this be logged?
+        :param save: Boolean should this be saved immediately?
+
+        This method brought to you via a grant from the Alfred P Sloan Foundation.
+        """
+        if self.has_coi == has_coi:
+            return
+
+        self.has_coi = has_coi
+
+        if log:
+            self.add_log(
+                action=PreprintLog.UPDATE_HAS_COI,
+                params={
+                    'user': auth.user._id,
+                    'value': self.has_coi
+                },
+                auth=auth,
+            )
+        if save:
+            self.save()
+
+    @rethrow_validation_error_for_serializer
+    def update_conflict_of_interest_statement(self, auth: tuple, coi_statement: str, log: bool = True, save: bool = True):
+        """
+        This method sets the `conflict_of_interest_statement` field for this preprint and logs that change.
+
+        :param auth: Tuple ('username', 'password')
+        :param coi_statement: String represents a user's conflict of interest statement for their preprint.
+        :param log: Boolean should this be logged?
+        :param save: Boolean should this be saved immediately?
+        :return:
+
+        This method brought to you via a grant from the Alfred P Sloan Foundation.
+        """
+        if self.conflict_of_interest_statement == coi_statement:
+            return
+
+        if not self.has_coi:
+            raise ValidationError('You do not have ability to edit a conflict of interest while the has_coi field is '
```

Comment:
`ValidationError` seems like the wrong exception -- this has nothing to do with django's validation

our own `PreprintStateError` seems more appropriate